### PR TITLE
Add some extra logging around outputs

### PR DESCRIPTION
**Pull request by agenticarus**
We're seeing some weird behaviour where some files are coming out with 0 bytes, or just not being generated at all. This adds some more debug logging to help us track down where/why.

### CODE REVIEW

File: src/build/build_step.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func buildTarget(tid int, state *core.BuildState, target *core.BuildTarget, runR
 	var postBuildOutput string
 	if state.PrepareOnly && state.IsOriginalTarget(target.Label) {
 		if target.IsFilegroup {
-			return fmt.Errorf("Can't prepare temporary directory for %s; filegroups don't have temporary directories", target.Label)
+			return fmt.Errorf("can't prepare temporary directory for %s; filegroups don't have temporary directories", target.Label)
```
Comment (peterebden) [inline src/build/build_step.go]: ytho?

File: src/build/build_step.go [hunk 2]
```diff
@@ -393,12 +393,17 @@ func moveOutputs(state *core.BuildState, target *core.BuildTarget) ([]string, bo
...
 		realOutput := path.Join(outDir, output)
 		if !core.PathExists(tmpOutput) {
-			return nil, true, fmt.Errorf("Rule %s failed to create output %s", target.Label, tmpOutput)
+			return nil, true, fmt.Errorf("rule %s failed to create output %s", target.Label, tmpOutput)
 		}
 		outputChanged, err := moveOutput(state, target, tmpOutput, realOutput)
 		if err != nil {
 			return nil, true, err
 		}
+		tmpStat, tmpErr := os.Stat(tmpOutput)
```
Comment (peterebden) [inline src/build/build_step.go]: can you add a TODO to remove this once the investigation is complete please?

File: src/build/build_step.go (hunk 1 above)
Comment (agenticarus) [reply src/build/build_step.go]: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: src/build/build_step.go (hunk 2 above)
Comment (agenticarus) [reply src/build/build_step.go]: Yeah, sure

**Pull request merged**